src/expressions/method-call-expr.md

```diff
@@ -15,22 +15,155 @@ let log_pi = pi.unwrap_or(1.0).log(2.72);
 
 When looking up a method call, the receiver may be automatically dereferenced or borrowed in order to call a method.
 This requires a more complex lookup process than for other functions, since there may be a number of possible methods to call.
-The following procedure is used:
 
-The first step is to build a list of candidate receiver types.
-Obtain these by repeatedly [dereferencing][dereference] the receiver expression's type, adding each type encountered to the list, then finally attempting an [unsized coercion] at the end, and adding the result type if that is successful.
-Then, for each candidate `T`, add `&T` and `&mut T` to the list immediately after `T`.
+The following procedure is used:
 
-For instance, if the receiver has type `Box<[i32;2]>`, then the candidate types will be `Box<[i32;2]>`, `&Box<[i32;2]>`, `&mut Box<[i32;2]>`, `[i32; 2]` (by dereferencing), `&[i32; 2]`, `&mut [i32; 2]`, `[i32]` (by unsized coercion), `&[i32]`, and finally `&mut [i32]`.
+## Determining candidate types
+
+First, a list of "candidate types" is assembled.
+
+These types are found by taking the receiver type and iterating, following either:
+
+* The built-in [dereference]; or
+* `<T as Receiver>::Target`
+
+to the next type. (If a step involved following the `Receiver` target, we also
+note whether it would have been reachable by following `<T as Deref::Target>`
+- this information is used later).
```

Comment:
```suggestion
note whether it would have been reachable by following `<T as
Deref::Target>` - this information is used later).
```

Github renders this as a bullet

Reply:
We'll want to unwrap all these lines before merging anyway.